### PR TITLE
Create firebase user if not exists

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* When a user tries to reset their password for an account with no firebase, re-create the firebase user.
    * We've seen this a few times for users who are invited, but never accept the invitation and try to log-in much later
    *  I believe this behavior was lost in https://github.com/google/exposure-notifications-verification-server/pull/902

### CODE REVIEW

File: pkg/controller/login/reset_password.go [hunk 1]
```diff
@@ -73,8 +74,14 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 				return
 			}
 
-			controller.InternalError(w, r, c.h, err)
-			return
+			// Re-create the firebase user if it does not exist.
+			// We have seen this disappear if a user never logs-on after being invited
+			if created, _ := c.authProvider.CreateUser(ctx, user.Name, user.Email, "", false, nil); created {
+				stats.Record(ctx, controller.MFirebaseRecreates.M(1))
```
Comment (sethvargo) [inline pkg/controller/login/reset_password.go:80]: This will fire in dev, so I don't think it's what we want. I'll add something to the other PR

Comment (mikehelmick) [reply pkg/controller/login/reset_password.go:80]: we probably don't want an alert on single instances, but if we see a rash over these over a time period, that seems noteworthy

Comment (whaught) [reply pkg/controller/login/reset_password.go:80]: I don't think we alert on this metric?

Comment (whaught) [reply pkg/controller/login/reset_password.go:80]: but also feel free to cannibalize this change

Comment (sethvargo) [reply pkg/controller/login/reset_password.go:80]: I see. We should move this into the login controller's metrics then. I can do that as part of the other PR if that's easier.

Comment (sethvargo) [reply pkg/controller/login/reset_password.go:80]: Pulled into https://github.com/google/exposure-notifications-verification-server/pull/1606

/close

**Pull request closed without merging**